src/_pytest/capture.py

```diff
@@ -685,11 +685,25 @@ def done(self):
         self._state = "done"
 
     def suspend(self):
+        if (
+            six.PY2
+            and self.name in ("stdout", "stderr")
+            and getattr(sys, "__{}__".format(self.name)) is self._old
+            and not hasattr(self, "_fdopen_done")
+            and not sys.platform.startswith("win32")
+        ):
+            self._fdopen_done = True
+            try:
+                # Ensure fd is unbuffered (#5134).
+                self._old = os.fdopen(self._old.fileno(), "wb+", 0)
+            except (UnsupportedOperation, OSError):
+                pass
```

Comment:
Note that `_py36_windowsconsoleio_workaround` does something similar already: https://github.com/pytest-dev/pytest/blob/f7ecd7243ccf708008a69b8156a0eeadaad2dad6/src/_pytest/capture.py#L804-L816